apps/client/src/app/util/validation.util.ts

```diff
@@ -0,0 +1,27 @@
+import { FormGroup } from '@angular/forms';
+import { plainToInstance } from 'class-transformer';
+import { validate } from 'class-validator';
+
+export function validateObjectForForm<T>(
+  obj: T,
+  ctor: { new (): T },
+  form: FormGroup,
+  onSuccess: () => void
+): void {
+  const objInstance = plainToInstance(ctor, obj);
+
+  validate(objInstance as object).then((errors) => {
+    if (errors.length === 0) {
+      onSuccess();
+    } else {
+      errors.forEach((error) => {
```

Comment:
`for (const error of errors) {` (to reduce nesting)

Reply:
Changed